develop/plugin_tutorial.rst

```diff
@@ -158,302 +104,231 @@ special note of the section that looks like this::
-channel the response is sent in the channel, and if the command is issued in a
-private dialog the response is sent in a private dialog. The text we want to
-display is simply the next number from our RNG (self.rng). We get that number
-by calling the random function, and then we str it just to make sure it is a
-nice printable string.
-
-Lastly, all plugin commands must be 'wrap'ed. What the wrap function does is
-handle argument parsing for plugin commands in a very nice and very powerful
-way. With no arguments, we simply need to just wrap it. For more in-depth
-information on using wrap check out the wrap tutorial (The astute Python
-programmer may note that this is very much like a decorator, and that's
-precisely what it is. However, we developed this before decorators existed and
-haven't changed the syntax due to our earlier requirement to stay compatible
-with Python 2.3.  As we now require Python 2.4 or greater, this may eventually
-change to support work via decorators.)
+``plugin.py`` includes the core code for the plugin. For most plugins this will
+include command handlers, as well as anything else that's relevant to its
+particular use case (database queries,
+:ref:`HTTP server endpoints <http_plugins>`,
+:ref:`IRC command triggers <do-method-handlers>`, etc.)
+
+As with any Python module, you'll need to import any dependencies you want,
+in addition to the standard ``supybot`` imports included in the plugin
+template::
+
+    import random
+
+The bulk of the plugin definition then resides in a subclass of
+:class:`callbacks.Plugin`. By convention, the class name is equal to the name of
+the plugin, though this is not strictly required (the actual linkage is done by
+the ``Class = Random`` statement at the end of the file). It is helpful to fill
+in the plugin docstring with some more details on how to actually use the plugin
+too: this info can be shown on a live bot using the
+``plugin help <plugin name>`` command.
+
+::
+
+    class Random(callbacks.Plugin):
+        """This plugin contains commands relating to random numbers, including random sampling from a list and a simple dice roller."""
+
+        def __init__(self, irc):
+            # Make sure to call the superclass' constructor when you define a custom one
+            super().__init__(irc)
+            self.rng = random.Random()  # create our rng
+            self.rng.seed() # automatically seeds with current time
+
+For this sample plugin, we define a custom constructor (``__init__``) that
+instantiates a random number generator instance and pre-seeds it. This isn't
+technically necessary for Python's ``random`` module, but this helps outline
+how to write a similar constructor. Notice in particular how you must pass in
+the ``irc`` argument in addition to ``self``.
+
+.. note::
+    TODO(jlu): semantically, what does ``irc`` refer to? Most plugins don't
+    actually reference it on load time.
```

Comment:
`Irc` object of the first network the bot connects to (on startup) or the network from which the plugin is loaded (if `@load`)